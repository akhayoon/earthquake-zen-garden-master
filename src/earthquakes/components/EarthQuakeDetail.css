.detail-container {
  margin: 0 auto;
  max-width: 25rem;
  white-space: nowrap;
}

.detail-container .field {
  padding-left: 0;
}