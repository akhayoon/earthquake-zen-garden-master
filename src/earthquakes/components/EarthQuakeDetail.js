/* eslint-disable arrow-body-style */
import React from 'react';
import { useParams } from 'react-router-dom';
import earthQuakeData from '../../data/earthquakes.json';
import Field from '../../common/components/Fields';
import formatTime from '../../common/utils/formatTime';
import './EarthQuakeDetail.css';

export default function EarthQuakeDetail() {
  const { id } = useParams();
  const detail = earthQuakeData.data.features.find((earthquake) => earthquake.id === id);

  return (
    <div className="detail-container">
      <h3>{detail.properties.place}</h3>
      <Field label="Title" value={detail.properties.title} />
      <Field label="Magnititude" value={detail.properties.mag} />
      <Field label="Time" value={formatTime(detail.properties.time)} />
      <Field label="Status" value={detail.properties.status} />
      <Field label="Tsunami" value={detail.properties.tsunami} />
      <Field label="Type" value={detail.properties.type} />
    </div>
  );
}
