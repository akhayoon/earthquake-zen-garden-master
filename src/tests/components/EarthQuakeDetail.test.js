import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, test } from '@jest/globals';
import renderer from 'react-test-renderer';
import EarthQuakeDetail from '../../earthquakes/components/EarthQuakeDetail';

// Mock react-router-dom in order to insert mock for useParams
// eslint-disable-next-line no-undef
jest.mock('react-router-dom', () => {
  return {
    // eslint-disable-next-line no-undef
    ...jest.requireActual('react-router-dom'),
    useParams: () => ({
      id: 'nc72999021',
    }),
  };
});

describe('Test EarthQuakeDetail Component', () => {
  test('it should match Snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <EarthQuakeDetail />
        </MemoryRouter>
        ,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
