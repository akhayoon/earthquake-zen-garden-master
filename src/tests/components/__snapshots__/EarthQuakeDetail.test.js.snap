// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test EarthQuakeDetail Component it should match Snapshot 1`] = `
<div
  className="detail-container"
>
  <h3>
    3km ENE of The Geysers, CA
  </h3>
  <div
    className="row"
  >
    <div
      className="field title"
    >
      Title
    </div>
    <div
      className="field value"
    >
      M 1.2 - 3km ENE of The Geysers, CA
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="field title"
    >
      Magnititude
    </div>
    <div
      className="field value"
    >
      1.19
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="field title"
    >
      Time
    </div>
    <div
      className="field value"
    >
      Apr 13, 2018 19:25 PM
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="field title"
    >
      Status
    </div>
    <div
      className="field value"
    >
      automatic
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="field title"
    >
      Tsunami
    </div>
    <div
      className="field value"
    >
      0
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="field title"
    >
      Type
    </div>
    <div
      className="field value"
    >
      earthquake
    </div>
  </div>
</div>
`;
